x-pack/plugins/security_solution/public/common/components/alerts_treemap/index.tsx

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Datum, ElementClickListener, PartialTheme } from '@elastic/charts';
+import { Chart, Partition, PartitionLayout, Settings } from '@elastic/charts';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { isEmpty } from 'lodash/fp';
+import React, { useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+
+import { useTheme } from '../charts/common';
+import { DraggableLegend } from '../charts/draggable_legend';
+import type { LegendItem } from '../charts/draggable_legend_item';
+import type { AlertSearchResponse } from '../../../detections/containers/detection_engine/alerts/types';
+import { getRiskScorePalette, RISK_SCORE_STEPS } from './lib/chart_palette';
+import { getFlattenedBuckets } from './lib/flatten/get_flattened_buckets';
+import { getFlattenedLegendItems } from './lib/legend/get_flattened_legend_items';
+import {
+  getGroupByFieldsOnClick,
+  getMaxRiskSubAggregations,
+  getUpToMaxBuckets,
+  hasOptionalStackByField,
+} from './lib/helpers';
+import { getLayersMultiDimensional, getLayersOneDimension } from './lib/layers';
+import { getFirstGroupLegendItems } from './lib/legend';
+import { NoData } from './no_data';
+import type { AlertsTreeMapAggregation, FlattenedBucket, RawBucket } from './types';
+
+export const DEFAULT_MIN_CHART_HEIGHT = 370; // px
+const DEFAULT_LEGEND_WIDTH = 300; // px
+
+export interface Props {
+  addFilter?: ({ field, value }: { field: string; value: string | number }) => void;
+  data: AlertSearchResponse<unknown, AlertsTreeMapAggregation>;
+  maxBuckets: number;
+  minChartHeight?: number;
+  stackByField0: string;
+  stackByField1: string | undefined;
+}
+
+const Wrapper = styled.div`
+  margin-top: -${({ theme }) => theme.eui.euiSizeS};
+`;
```

Comment:
You could also use `<EuiSpacer size="s" />` instead and wrap the component in a regular `div`.